src/ranged.cpp

```diff
@@ -141,6 +141,13 @@ dealt_projectile_attack Creature::projectile_attack( const projectile &proj_arg,
     const bool no_item_damage = proj_effects.count( "NO_ITEM_DAMAGE" ) > 0;
     const bool do_draw_line = proj_effects.count( "DRAW_AS_LINE" ) > 0;
     const bool null_source = proj_effects.count( "NULL_SOURCE" ) > 0;
+    // Determines whether it can penetrate obstacles
+    const bool is_bullet = proj_arg.speed >= 200 && std::any_of( proj_arg.impact.damage_units.begin(),
+                           proj_arg.impact.damage_units.end(),
+                           []( const damage_unit &dam )
+    {
+        return dam.type == DT_CUT;
```

Comment:
IRL, a rock that's badly aimed, weakly thrown, or thrown from far away might not go through a window. With 2 throwing, 5 strength, 5 dexterity, and from 1 tile away, 4 out of 14 throws broke a window. With 2 throwing, 8 strength, and 8 dexterity, you can reliably break it from 3 tiles away. With 0 throwing, 8 strength, and 8 dexterity, you can reliably break a window from 2 tiles away. With 1 throwing, 5 strength, and 5 dexterity, it's very difficult (but possible) to break a window from 1 tile away.